frontend/src/components/MeetingRoom/EmojiGrid/EmojiGrid.tsx

```diff
@@ -1,76 +1,56 @@
-import { Grid, Grow, Paper, Popper, Tooltip } from '@mui/material';
-import { EmojiEmotions } from '@mui/icons-material';
-import { ReactElement, useRef, useState } from 'react';
-import { ClickAwayListener, PopperChildrenProps } from '@mui/base';
-import ToolbarButton from '../ToolbarButton';
-import emojiMap from '../../../utils/emojis';
-import SendEmojiButton from '../SendEmojiButton';
-import displayOnDesktop from '../../../utils/displayOnDesktop';
+import { Dispatch, ReactElement, RefObject, SetStateAction } from 'react';
+import useIsSmallViewport from '../../../hooks/useIsSmallViewport';
+import EmojiGridMobile from './EmojiGridMobile';
+import EmojiGridDesktop from './EmojiGridDesktop';
+
+export type EmojiGridProps = {
+  isEmojiGridOpen: boolean;
+  setIsEmojiGridOpen: Dispatch<SetStateAction<boolean>>;
+  anchorRef: RefObject<HTMLButtonElement | null>;
+  isParentOpen: boolean;
+};
 
 /**
  * EmojiGrid Component
  *
- * Displays a clickable button that opens a grid of emojis.
+ * Displays a grid of emojis.
+ * @param {EmojiGridProps} props - the props for the component
+ *  @property {RefObject<HTMLButtonElement | null>} anchorRef - the button ref for the grid
+ *  @property {boolean} isEmojiGridOpen - whether the component will be open initially
+ *  @property {Dispatch<SetStateAction<boolean>>} setIsEmojiGridOpen - toggle whether the emoji grid is shown or hidden
+ *  @property {boolean} isParentOpen - whether the ToolbarOverflowMenu is open
  * @returns {ReactElement} - The EmojiGrid Component.
  */
-const EmojiGrid = (): ReactElement => {
-  const anchorRef = useRef<HTMLButtonElement>(null);
-  const [open, setOpen] = useState<boolean>(false);
+const EmojiGrid = ({
+  anchorRef,
+  isEmojiGridOpen,
+  setIsEmojiGridOpen,
+  isParentOpen,
+}: EmojiGridProps): ReactElement => {
+  const isSmallViewport = useIsSmallViewport();
 
-  const handleClose = () => {
-    setOpen(false);
-  };
+  const handleClickAway = (event: MouseEvent | TouchEvent) => {
+    const target = event.target as HTMLElement;
 
-  const handleToggle = () => {
-    setOpen((prevOpen) => !prevOpen);
+    if (isSmallViewport && !target.closest('#emoji-grid-toggle')) {
+      return;
+    }
+    // If a user clicks the toggle button, we save their preference for later
```

Comment:
I think this comment and the code above could be clearer.

Something like: If user didn't click emoji toggle button - do nothing as this will be handled by toolbar clickAway listner

Otherwise if user clicked toggle button then close the emoji drawer.